Content.Shared/Chat/V2/Systems/Messages.cs

```diff
@@ -0,0 +1,285 @@
+/// <param name="message">The message, possibly altered from when it was sent</param>
+/// <param name="id">The round-unique ID of the message</param>
+[Serializable, NetSerializable]
+public abstract class ChatSuccessEvent(ChatContext context, NetEntity speaker, string asName, string message, uint id) : EntityEventArgs
+{
+    public ChatContext Context = context;
+    public NetEntity Speaker = speaker;
+    public string AsName = asName;
+    public readonly string Message = message;
+    public uint Id = id;
+}
+
+/// <summary>
+/// Raised to inform clients that an entity has spoken.
+/// </summary>
+[Serializable, NetSerializable]
+public sealed class VerbalChatEvent(
+    ChatContext context,
+    NetEntity speaker,
+    string asName,
+    string message,
+    uint id,
+    VerbalChatChannel chatChannel
+) : ChatSuccessEvent(context, speaker, asName, message, id)
+{
+    public VerbalChatChannel ChatChannel = chatChannel;
+}
+
+/// <summary>
+/// Raised when a mob emotes.
+/// </summary>
+/// <param name="speaker">The speaker of the message</param>
+/// <param name="asName">What name the speaker should present as</param>
+/// <param name="message">The message, possibly altered from when it was sent</param>
+/// <param name="id">The round-unique ID of the message</param>
+[Serializable, NetSerializable]
+public sealed class VisualChatEvent(
+    ChatContext context,
+    NetEntity speaker,
+    string asName,
+    string message,
+    uint id,
+    VisualChatChannel chatChannel
+) : ChatSuccessEvent(context, speaker, asName, message, id)
+{
+    public VisualChatChannel ChatChannel = chatChannel;
+}
+
+/// <summary>
+/// Raised when an announcement is made.
+/// </summary>
+[Serializable, NetSerializable]
+public sealed class AnnouncementEvent(
+    ChatContext context,
+    string asName,
+    string message,
+    uint id,
+    Color? messageColorOverride = null
+) : ChatSuccessEvent(context, NetEntity.Invalid, asName, message, id)
+{
+    public Color? MessageColorOverride = messageColorOverride;
+}
+
+public sealed class OutOfCharacterChatEvent(
+    ChatContext context,
+    NetEntity speaker,
+    string asName,
+    string message,
+    uint id,
+    OutOfCharacterChatChannel chatChannel
+) : ChatSuccessEvent(context, speaker, asName, message, id)
+{
+    public OutOfCharacterChatChannel ChatChannel = chatChannel;
+}
+
+#endregion
+
+#region Server-Generated Chat Events
+
+/// <summary>
+/// Raised when a character speaks on a radio channel.
+/// </summary>
+/// <param name="speaker">The speaker of the message</param>
+/// <param name="asName">What name the speaker should present as</param>
+/// <param name="message">The message, possibly altered from when it was sent</param>
+/// <param name="id">The round-unique ID of the message</param>
+/// <param name="channel">The channel the message is on</param>
+[Serializable, NetSerializable]
+public sealed class RadioEvent(
```

Comment:
This is how Radio is proposed to work: you whisper and provide a radio channel as the frontend. We make sure that's legal in the backend. Then a handler will go "oh, this created event has a radio channel, I'll send a radio message!" and that gets handled in the SAME RADIO CODE AS INTERCOMS ET AL.

No more stupid "radio channel" nonsense. You send by talking, your radio receives audio. The server figures out how to go from talking to radio. One system, not like sixteen.